source/exe/main_common.h

```diff
@@ -21,30 +23,28 @@ class ProdComponentFactory : public Server::ComponentFactory {
                                    Server::Configuration::Initial& config) override;
 };
 
+struct AdminResponse {
```

Comment:
Arguably, where we want to go in the future with this is to have the admin endpoint itself a proto defined API, where each endpoint exports out some proto object. Then, we would not be using strings/headers here, but instead have a wrapper proto object we would hand to the caller. This isn't the PR for that, but maybe you could add some TODOs.

Reply:
Good point - will do.